spinalcordtoolbox/deepseg_sc/core.py

```diff
@@ -548,7 +548,7 @@ def deep_segmentation_spinalcord(im_image, contrast_type, ctr_algo='cnn', ctr_fi
     # Binarize the resampled image (except for soft segmentation, defined by threshold_seg=-1)
     if threshold_seg >= 0:
         logger.info("Binarizing the resampled segmentation...")
-        im_seg_r.data = im_seg_r.data.astype(np.uint8)
+        im_seg_r.data = (im_seg_r.data > 0.5).astype(np.uint8)
```

Comment:
Sorry to change my mind but: was thinking of setting 0 instead of 0.5 so that "no voxel are threshold" during this step and everything is controlled by the "optimised threshold" just after the prediction (ie "everything is controlled by only one threshold"). Does that make sense? or it does not really matter?

Reply:
actually, to have a proper (i.e. unbiased) thresholding, after resampling, it should be set to 0.5 (equivalent to nearest neighbor). Setting to 0 will introduce a bias.

Reply:
Right! Thanks for clarifying, makes sense!